docs/explanation/lifecycle-layer.rst

```diff
@@ -0,0 +1,207 @@
+
+From lifecycle to OCI layer
+===========================
+
+Rockcraft is a tool that creates OCI images using the same concepts and
+mechanisms that create snaps and charms: the lifecycle language from
+Craft Parts. There is a significant difference between the works of the Craft
+lifecycle and the OCI specification, and one of Rockcraft's jobs is to bridge
+the gap between these two worlds. This page describes how this is accomplished.
+
+.. note::
+   It is not necessary to know these details to use the tool effectively, but
+   they might illuminate some concepts and help understand *why* the contents of
+   a given rock are the way they are.
```

Comment:
this is superfluous as you shouldn't need to read any explanation documentation for correct use of the tool

Reply:
I disagree with most of the comments in this document about it being too much, especially if I go and read about explanation on https://diataxis.fr/explanation/

That said, this clarification is essentially what explanation can cover which is why I say it is superfluous.

Reply:
> this is superfluous as you shouldn't need to read any explanation documentation for correct use of the tool

I don't think this is as cut-and-dry as you suggest, but more importantly I don't think a given user *knows* that the explanations are not necessary read. I imagine that many Rockcraft users will not be familiar with Diatáxis, and in my opinion they shouldn't have to be.

Reply:
@sergiusens the comments are not about the pertinence of this explanation, but rather its place. Again referring to https://github.com/canonical/rockcraft/pull/445#discussion_r1423913874, we are missing an entry-level explanation to OCI Layering for rocks (which btw, is a common question). Then, yes, this page, would become an in-depth developer-oriented explanation of what happens when priming contents to an OCI layer.